notifiers/providers/sendgrid.py

```diff
@@ -0,0 +1,623 @@
+                            'enable': {
+                                'type': 'boolean',
+                                'title': 'Indicates if this setting is enabled'
+                            },
+                            'utm_source': {
+                                'type': 'string',
+                                'title': 'Name of the referrer source.' +
+                                         ' (e.g. Google, SomeDomain.com,' +
+                                         ' or Marketing Email)'
+                            },
+                            'utm_medium': {
+                                'type': 'string',
+                                'title': 'Name of the marketing medium. ' +
+                                         '(e.g. Email)'
+                            },
+                            'utm_term': {
+                                'type': 'string',
+                                'title': 'Used to identify any paid keywords.'
+                            },
+                            'utm_content': {
+                                'type': 'string',
+                                'title': 'Used to differentiate your ' +
+                                         'campaign from advertisements.'
+                            },
+                            'utm_campaign': {
+                                'type': 'string',
+                                'title': 'The name of the campaign'
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        'additionalProperties': False
+    }
+
+    def _prepare_data(self, data: dict) -> dict:
+        if data.get('from_'):
+            data['from'] = data.pop('from_')
+
+        # supporting a bare string 'from', this simplifies
+        # the API and allows us to use the CLI
+        if isinstance(data['from'], str):
+            data['from'] = {
+                'email': data.pop('from')
+            }
+
+        if data.get('message'):
+            content = [
+                {
+                    'type': 'text/plain',
+                    'value': data.pop('message')
+                }
+            ]
+            data['content'] = content
+
+        # inserting support for a 'to' argument, since sendgrid is a little
+        # complicated about setting that
+        if data.get('to'):
+
+            if not data.get('personalizations'):
+                data['personalizations'] = []
+
+            data['personalizations'].append(
+                {
+                    'to': [
+                        {
+                            'email': data.pop('to')
+                        }
+                    ]
+                }
+            )
+
+
+        return data
+
+    def _send_notification(self, data: dict) -> Response:
+        headers = {
+            'Authorization': f'Bearer {data["api_key"]}',
+            'Content-Type': 'application/json'
```

Comment:
no need to set content type here, request sets it to json when you use the json keyword to send data